plugins/modules/dcnm_vrf.py

```diff
@@ -342,15 +404,40 @@ def diff_for_attach_deploy(self, want_a, have_a):
             if have_a:
                 for have in have_a:
                     if want['serialNumber'] == have['serialNumber']:
-                        found = True
+                        if want['extensionValues'] != "" and have['extensionValues'] != "":
+                            want_ext_values = want['extensionValues']
+                            want_ext_values = eval(want_ext_values)
+                            have_ext_values = have['extensionValues']
+                            have_ext_values = eval(have_ext_values)
+
+                            want_e = eval(want_ext_values['VRF_LITE_CONN'])
+                            have_e = eval(have_ext_values['VRF_LITE_CONN'])
+
+                            if want_e['VRF_LITE_CONN'][0]['IF_NAME'] == have_e['VRF_LITE_CONN'][0]['IF_NAME']:
```

Comment:
Given that both `want_e[key]0[key]` and `have_e[key]0[key]` have the same keys, we should be able to significantly simplify this logic and just do a dict compare:

For example:

```python
>>> want_e
{'VRF_LITE_CONN': [{'DOT1Q_ID': '429', 'IP_MASK': '255.255.255.0', 'NEIGHBOR_IP': '192.168.5.5'}]}
>>> have_e
{'VRF_LITE_CONN': [{'DOT1Q_ID': '429', 'IP_MASK': '255.255.255.0', 'NEIGHBOR_IP': '192.168.5.5'}]}
>>> 
>>> want_e == have_e
True
>>> 
>>> have_e['VRF_LITE_CONN'][0]['DOT1Q_ID'] = '500'
>>> 
>>> want_e
{'VRF_LITE_CONN': [{'DOT1Q_ID': '429', 'IP_MASK': '255.255.255.0', 'NEIGHBOR_IP': '192.168.5.5'}]}
>>> have_e
{'VRF_LITE_CONN': [{'DOT1Q_ID': '500', 'IP_MASK': '255.255.255.0', 'NEIGHBOR_IP': '192.168.5.5'}]}
>>> 
>>> want_e == have_e
False
>>> 
```
>>> want_e
{'VRF_LITE_CONN': [{'DOT1Q_ID': '429', 'IP_MASK': '255.255.255.0', 'NEIGHBOR_IP': '192.168.5.5'}]}
>>> have_e
{'VRF_LITE_CONN': [{'DOT1Q_ID': '429', 'IP_MASK': '255.255.255.0', 'NEIGHBOR_IP': '192.168.5.5', 'EXTRA_KEY': 'foo'}]}
>>> want_e == have_e
False
>>> 


This won't work for complex dict compares but in this case it should.

Reply:
Done.

Reply:
Tried this for all the IT cases, i see some idempotence cases are failing due to this change, reverted to the original.

Reply:
Ok for now.